Backbone.Tests.ArchUnit/DomainEvents.cs

```diff
@@ -0,0 +1,57 @@
+﻿using ArchUnitNET.Domain.Extensions;
+using ArchUnitNET.Fluent.Conditions;
+using ArchUnitNET.xUnit;
+using Backbone.BuildingBlocks.Domain.Events;
+using Backbone.Tooling.Extensions;
+using static ArchUnitNET.Fluent.ArchRuleDefinition;
+
+namespace Backbone.Backbone.Tests.ArchUnit;
+
+public class DomainEvents
+{
+    [Fact]
+    public void IncomingDomainEventsShouldNotHaveConstructors()
+    {
+        Types()
+            .That().AreAssignableTo(typeof(DomainEvent))
+            .And().ResideInNamespace("Backbone.Modules.*.Domain.DomainEvents.Incoming.*", true)
```

Comment:
I think we might as well just check for ".*Incoming.*". I don't think there will ever be and outgoing domain event that is in a namespace containing "Incoming". :-)